Doc/library/typing.rst

```diff
@@ -1266,213 +1322,320 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
    .. versionadded:: 3.11
 
-Building generic types
-""""""""""""""""""""""
+Building generic types and type aliases
+"""""""""""""""""""""""""""""""""""""""
 
-These are not used in annotations. They are building blocks for creating generic types.
+The following objects are not used directly in annotations. Instead, they are building blocks
+for creating generic types and type aliases.
+They can be created through special syntax (:ref:`type parameter lists <type-params>`
+and the :keyword:`type` statement). For compatibility with Python 3.11 and earlier, they
+can also be created without the dedicated syntax, as documented below.
 
 .. class:: Generic
 
    Abstract base class for generic types.
 
-   A generic type is typically declared by inheriting from an
-   instantiation of this class with one or more type variables.
-   For example, a generic mapping type might be defined as::
+   A generic type is typically declared by adding a list of type parameters
+   after the class name::
 
-      class Mapping(Generic[KT, VT]):
+      class Mapping[KT, VT]:
           def __getitem__(self, key: KT) -> VT:
               ...
               # Etc.
 
-   This class can then be used as follows::
+   Such a class implicitly inherits from ``Generic``.
+   The runtime semantics of this syntax are discussed in the
+   :ref:`Language Reference <generic-classes>`.
 
-      X = TypeVar('X')
-      Y = TypeVar('Y')
+   This class can then be used as follows::
 
-      def lookup_name(mapping: Mapping[X, Y], key: X, default: Y) -> Y:
+      def lookup_name[X, Y](mapping: Mapping[X, Y], key: X, default: Y) -> Y:
           try:
               return mapping[key]
           except KeyError:
               return default
 
-.. class:: TypeVar
+   Generic classes can also be declared by explicitly inheriting from
+   ``Generic``. In this case, the type parameters must be created
+   separately::
+
+      KT = TypeVar('KT')
+      VT = TypeVar('VT')
+
+      class Mapping(Generic[KT, VT]):
+          def __getitem__(self, key: KT) -> VT:
+              ...
+              # Etc.
+
+.. class:: TypeVar(name, *constraints, bound=None, covariant=False, contravariant=False, infer_variance=False)
```

Comment:
I worry that people scanning the doc will see the signature and assume that `infer_variance` defaults to `False` for PEP-695 `TypeVar`s as well. Could we have a comment slightly higher up stating that type checkers are expected to automatically infer the variance for PEP-695 TypeVars? (I know you go into it lower down, but my worry is that it's too far away from the signature here.)

Reply:
Moved the variance paragraph up a bit. I'm not sure it should go too far up, since for the most part people shouldn't worry about variance.